services/datamanager/builtin/sync_test.go

```diff
@@ -289,6 +303,10 @@ func TestDataCaptureUpload(t *testing.T) {
 			}
 			failedURs := mockService.getFailedDCUploadRequests()
 
+			if tc.emptyFile {
```

Comment:
Is this not captured by the check below already? Since len(capturedData) should equal 0 in this case (based on above)

Reply:
Good catch!